cli/assets/templates/zig/build.zig

```diff
@@ -1,42 +1,66 @@
-            const number = try std.fmt.parseUnsigned(u32, pre[4..], 10);
-            if (number >= 258) return;
-        } else return;
+            // Merged here: https://github.com/ziglang/zig/pull/10572
+            return std.mem.startsWith(u8, pre, "dev.") and zig_version.order(try std.SemanticVersion.parse("0.10.0-dev.258")).compare(.gte);
+        }
     }
+    // Backported here: https://github.com/ziglang/zig/commit/6f49233ac6a6569b909b689f22fc260dc8c19234
+    return zig_version.order(try std.SemanticVersion.parse("0.9.1")).compare(.gte);
+}
+
+test "stack version check" {
+    const expect = std.testing.expect;
+    const parse = std.SemanticVersion.parse;
+    try expect(!try version_supports_stack_first(try parse("0.8.0")));
+
+    try expect(!try version_supports_stack_first(try parse("0.9.0")));
+    try expect(!try version_supports_stack_first(try parse("0.9.1-dev.259")));
+    try expect(try version_supports_stack_first(try parse("0.9.1")));
 
-    if (zig_version.minor == 9
-        and zig_version.pre == null
-        and zig_version.patch >= 1) return;
-    
-    // `--stack-first` option have been reenabled on wasm targets with https://github.com/ziglang/zig/pull/10572
-    std.debug.panic("0.9.1 or 0.10.0-dev.258 are minimal required versions of zig (got {})", .{zig_version});
+    // Conservatively don't recognize tags other than 'dev'.
+    try expect(!try version_supports_stack_first(try parse("0.10.0-aev.259")));
+    try expect(!try version_supports_stack_first(try parse("0.10.0-zev.259")));
+
+    try expect(!try version_supports_stack_first(try parse("0.10.0-dev.257")));
+    try expect(try version_supports_stack_first(try parse("0.10.0-dev.258")));
+    try expect(try version_supports_stack_first(try parse("0.10.0-dev.259")));
+    try expect(try version_supports_stack_first(try parse("0.10.0")));
+
+    try expect(try version_supports_stack_first(try parse("0.10.1-dev.100")));
+    try expect(try version_supports_stack_first(try parse("0.10.1-dev.300")));
+    try expect(try version_supports_stack_first(try parse("0.10.1")));
+
+    try expect(try version_supports_stack_first(try parse("1.0.0")));
 }
 
 pub fn build(b: *std.build.Builder) !void {
-    try check_zig_version();
-
+    const zig_version = @import("builtin").zig_version;
     const mode = b.standardReleaseOptions();
     const lib = b.addSharedLibrary("cart", "src/main.zig", .unversioned);
     lib.setBuildMode(mode);
     lib.setTarget(.{ .cpu_arch = .wasm32, .os_tag = .freestanding });
     lib.import_memory = true;
     lib.initial_memory = 65536;
     lib.max_memory = 65536;
-    lib.stack_size = 14752;
+    if (try version_supports_stack_first(zig_version)) {
+        lib.stack_size = 14752;
+    } else {
+        // `--stack-first` option have been reenabled on wasm targets with https://github.com/ziglang/zig/pull/10572
+        std.log.warn("Update to Zig >=0.9.1 or >=0.10.0-dev.258 to detect stack overflows at runtime.", .{});
```

Comment:
```suggestion
        std.log.warn("Update to Zig >=0.9.1 (or >=0.10.0-dev.258 for nightly) to detect stack overflows at runtime.", .{});
```

Reply:
I am not in the zig community, but i think `>=0.10.0-dev.258` would seem a bit too weird to a newbie.